service/Core/Handlers/SaveRecordsHandler.cs

```diff
@@ -45,6 +45,7 @@ private static string GetRecordId(string documentId, string partId)
 
     private readonly IPipelineOrchestrator _orchestrator;
     private readonly List<IMemoryDb> _memoryDbs;
+    private readonly KernelMemoryConfig _config;
```

Comment:
I'd avoid exposing the entire config, using something like `private readonly int _upsertBatchSize;` instead